File: Simplace/SimplaceReferenceETPriestleyTaylor/src/java/SimplaceReferenceETPriestleyTaylor/ReferenceETPriestleyTaylor_Component.java
public class ReferenceETPriestleyTaylor_Component
{
    
    public ReferenceETPriestleyTaylor_Component() { }

    ReferenceETPriestleyTaylor _ReferenceETPriestleyTaylor = new ReferenceETPriestleyTaylor();

    public double getcAlphaPT()
    { return _ReferenceETPriestleyTaylor.getcAlphaPT(); }
    public void setcAlphaPT(double _cAlphaPT){
    _ReferenceETPriestleyTaylor.setcAlphaPT(_cAlphaPT);
    }

    public double getcAltitude()
    { return _ReferenceETPriestleyTaylor.getcAltitude(); }
    public void setcAltitude(double _cAltitude){
    _ReferenceETPriestleyTaylor.setcAltitude(_cAltitude);
    }
    public void  Calculate_Model(ReferenceETPriestleyTaylor_State s, ReferenceETPriestleyTaylor_State s1, ReferenceETPriestleyTaylor_Rate r, ReferenceETPriestleyTaylor_Auxiliary a, ReferenceETPriestleyTaylor_Exogenous ex)
    {
        _ReferenceETPriestleyTaylor.Calculate_Model(s, s1, r, a, ex);
    }
    private double cAlphaPT;
    private double cAltitude;
    public ReferenceETPriestleyTaylor_Component(ReferenceETPriestleyTaylor_Component toCopy) // copy constructor 
    {
        this.cAlphaPT = toCopy.getcAlphaPT();
        this.cAltitude = toCopy.getcAltitude();

    }
}

File: Simplace/SimplaceReferenceETPriestleyTaylor/src/cs/SimplaceReferenceETPriestleyTaylor/ReferenceETPriestleyTaylor_Component.cs
public class ReferenceETPriestleyTaylor_Component
{
    
    /// <summary>
    /// Constructor of the ReferenceETPriestleyTaylor_Component component")
    /// </summary>  
    public ReferenceETPriestleyTaylor_Component() { }
    

    //Declaration of the associated strategies
    ReferenceETPriestleyTaylor _ReferenceETPriestleyTaylor = new ReferenceETPriestleyTaylor();

    public double cAlphaPT
    {
        get
        {
             return _ReferenceETPriestleyTaylor.cAlphaPT; 
        }
        set
        {
            _ReferenceETPriestleyTaylor.cAlphaPT = value;
        }
    }
    public double cAltitude
    {
        get
        {
             return _ReferenceETPriestleyTaylor.cAltitude; 
        }
        set
        {
            _ReferenceETPriestleyTaylor.cAltitude = value;
        }
    }

    public void  CalculateModel(ReferenceETPriestleyTaylor_State s, ReferenceETPriestleyTaylor_State s1, ReferenceETPriestleyTaylor_Rate r, ReferenceETPriestleyTaylor_Auxiliary a, ReferenceETPriestleyTaylor_Exogenous ex)
    {
        _ReferenceETPriestleyTaylor.CalculateModel(s,s1, r, a, ex);
    }
    
    public ReferenceETPriestleyTaylor_Component(ReferenceETPriestleyTaylor_Component toCopy): this() // copy constructor 
    {

        cAlphaPT = toCopy.cAlphaPT;
        cAltitude = toCopy.cAltitude;
    }
}